src/canvas-extractor/edfi_canvas_extractor/mapping/submissions.py

```diff
@@ -19,12 +19,14 @@ def _get_date_formated(date) -> str:
 
 
 def _get_status(row: pd.Series):
-    if row["late"] == 1:
+    if row["late"] is True:
         return "late"
-    if row["missing"] == 1:
+    if row["missing"] is True:
         return "missing"
-    if row["graded_at"] is not None:
+    if pd.isnull(row["graded_at"]) is False:
         return "graded"
+    if pd.isnull(row["SubmissionDateTime"]) is False:
+        return "upcoming"
```

Comment:
I created a new assignment in section 103 (ENG-1), with due date set to 2021-12-17. The output file has "on-time" for all of the records instead of "upcoming". Here's an example raw response from the GET request:

```javascript
{
        "assignment_id": 1221,
        "id": 11464,
        "grading_period_id": null,
        "body": null,
        "url": null,
        "grade": null,
        "score": null,
        "submitted_at": null,
        "user_id": 111,
        "submission_type": null,
        "workflow_state": "unsubmitted",
        "grade_matches_current_submission": true,
        "graded_at": null,
        "grader_id": null,
        "attempt": null,
        "cached_due_date": "2021-12-18T05:59:59Z",
        "excused": null,
        "late_policy_status": null,
        "points_deducted": null,
        "extra_attempts": null,
        "posted_at": null,
        "late": false,
        "missing": false,
        "seconds_late": 0,
        "entered_grade": null,
        "entered_score": null,
        "preview_url": "https://edfialliance.instructure.com/courses/104/assignments/1221/submissions/111?preview=1&version=0",
        "anonymous_id": "Lwc1w"
    },
```

In the same section, assignment 114 has some records like this one, and they are showing up with status "upcoming" instead of "missing". So I wonder if the change on line 24 was incorrect.

```javascript
    {
        "assignment_id": 114,
        "id": 11456,
        "grading_period_id": null,
        "body": null,
        "url": null,
        "grade": null,
        "score": null,
        "submitted_at": null,
        "user_id": 108,
        "submission_type": null,
        "workflow_state": "unsubmitted",
        "grade_matches_current_submission": true,
        "graded_at": null,
        "grader_id": null,
        "attempt": null,
        "cached_due_date": "2021-05-01T06:01:00Z",
        "excused": null,
        "late_policy_status": null,
        "points_deducted": null,
        "extra_attempts": null,
        "posted_at": null,
        "late": false,
        "missing": true,
        "seconds_late": 10510801,
        "entered_grade": null,
        "entered_score": null,
        "preview_url": "https://edfialliance.instructure.com/courses/104/assignments/114/submissions/108?preview=1&version=0",
        "anonymous_id": "yxUaK"
    }
```

Finally, I created my own Submission on the assignment above. The API response has `"late": true`, but the status in the extracted file is "upcoming" instead of "late."

Finally, shouldn't line 28 be looking for cases where the `SubmissionDateTime` _is null_ instead of _not is null_?